core/src/main/java/io/grpc/internal/AbstractServerImplBuilder.java

```diff
@@ -71,6 +72,11 @@ protected AbstractServerImplBuilder() {
   }
 
   @Override
+  public final T directExecutor() {
+    return executor(MoreExecutors.directExecutor());
```

Comment:
Makes sense. The `DirectExecutorService` uses a lock internally [1]. Will test if that also makes a difference.

[1] https://code.google.com/p/guava-libraries/source/browse/guava/src/com/google/common/util/concurrent/MoreExecutors.java?name=jdk5-backport-sync#280